components/_helpers.js

```diff
@@ -1,19 +1,21 @@
 const modSep = '_';
 
-function getClassNamesWithMods(baseClass = '', mods = {}) {
+function getClassNamesWithMods(baseClass = '', ...modsSet) {
   let classes = [].concat(baseClass);
   baseClass = classes[0];
 
-  if (mods instanceof Array) {
-    classes = classes
-      .concat(mods.filter(Boolean).map(mode => `${baseClass}${modSep}${mode}`));
-  } else {
-    Object.keys(mods).forEach((mode) => {
-      if (mods[mode]) {
-        classes.push(`${baseClass}${modSep}${mode}`);
-      }
-    });
-  }
+  modsSet.forEach((mods) => {
```

Comment:
On one side, this function will be called thousands of times by thousands of components on a page and it is important to make it as performant as possible (yes, you can call it "micro-optimization", but it is an optimization for frequently called function)
On the other side, I agree with you about readability.